### PR TITLE
Sync report names across ranks

### PR DESCRIPTION
**Pull request by jorblancoa**
- Fixed deadlock bug when some ranks don't have all the reports

### CODE REVIEW

**Comment by WeinaJi** [comment]
I have run my CI test with this PR and it solved the deadlock issue. Thanks @jorblancoa

**Review by WeinaJi — APPROVED** [review]
Looks good to me.

**Comment by jorblancoa** [comment]
Runing blueconfigs pipeline here
https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/48780

**Comment by pramodk** [comment]
@jorblancoa : let's not forget to update one of the existing BlueConfig to cover this regression :)

File: src/library/implementation_interface.hpp [hunk 1]
```diff
@@ -129,86 +198,33 @@ struct ParallelImplementation {
         int num_reports = report_names.size();
         MPI_Comm_split(MPI_COMM_WORLD, num_reports == 0, 0, &SonataReport::has_nodes_);
```
[the remark this answers is not in the document]
Comment (jorblancoa) [reply src/library/implementation_interface.hpp:199]: I have created the issue as good for beginners :)
https://github.com/BlueBrain/libsonatareport/issues/27

**Comment by WeinaJi** [comment]
> @jorblancoa : let's not forget to update one of the existing BlueConfig to cover this regression :)

The deadlock issue showed up in the blueconfig ngv test with Fernando's mulitpop improvements, not the main branch. 
Once this PR is merged, Fernando's MRs will pass and be merged.
https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/merge_requests/35
https://bbpgitlab.epfl.ch/hpc/sim/neurodamus-py/-/merge_requests/39

**Pull request merged**